R/utils.R

```diff
@@ -1,40 +1,274 @@
+#' @param I Matrix: defining the interaction between each pair of species
+#' @param pair Numeric: a vector with a length of 2, indicating the 2 focusing 
+#' species in the process of applying the interaction types
+#' @param interType Character: one of 'mutualism', 'commensalism', 'parasitism',
+#' 'amensalism', or 'competition'. Defining the interaction type
+#' @examples
+#' applyInteractionType(matrix(0, nrow = 3, ncol = 3), 
+#'                      pair = c(1,2), 
+#'                      interType = "parasitism")
+#' @return A matrix of interaction types with one pair changed
+#' @export
+applyInteractionType <- function(I, pair, interType){
+  if (rbinom(1,1,0.5)){
+    pair <- rev(pair)
+  }
+  if (interType=='mutualism'){
+    I[pair[1],pair[2]] <- 1
+    I[pair[2],pair[1]] <- 1
+    return(I)
+  }else if (interType=='commensalism'){
+    I[pair[1],pair[2]] <- 1
+    I[pair[2],pair[1]] <- 0
+    return(I)
+  }else if (interType=='parasitism'){
+    I[pair[1],pair[2]] <- 1
+    I[pair[2],pair[1]] <- -1
+    return(I)
+  }else if (interType=='amensalism'){
+    I[pair[1],pair[2]] <- 0
+    I[pair[2],pair[1]] <- -1
+    return(I)
+  }else if (interType=='competition'){
+    I[pair[1],pair[2]] <- -1
+    I[pair[2],pair[1]] <- -1
+    return(I)
+  }
+}
+
+#' @title generate interactions according to five types of interactions and
+#' their weights
+#' @param n_species Integer: defining the dimension of matrix of interaction
+#' @param weights Numeric: defining the weights of mutualism, commensalism, 
+#' parasitism, amensalism, and competition in all interspecies interactions.
+#' @param connectance Numeric: defining the density of the interaction network.
+#' Ranging from 0 to 1
+#' @examples
+#' I <- getInteractions(n_species = 10, 
+#'                      weights = c(1,1,1,1,1), 
+#'                      connectance = 0.5)
+#' @return A matrix of interactions with all interactions changed according to
+#' the weights and connectance.
+#' @export
+getInteractions <- function(n_species, weights, connectance){
+    I <- matrix(0, n_species, n_species)
+    interactions <- c('mutualism', 'commensalism', 'parasitism', 
+                      'amensalism', 'competition')
+    probs <- abs(weights)/sum(abs(weights))
+    combinations <- combn(n_species, 2)
+    for (i in sample(seq_along(combinations[1,]),as.integer(ncol(combinations)*connectance), replace=FALSE)){
+        I <- applyInteractionType(I, combinations[,i], sample(interactions, 1, prob = probs))
+    }
+    return (I)
+}
+
+#' @title only get the maximum value from a row 
+#' @description A helper function to be used in combination with apply()
+#' @param aRow Vector: a row in a matrix/dataframe
+#' @return A vector of numbers with only the first maximun value kept, and other
+#' numbers changed to 0
+#' @export
+getRowMax <- function(aRow){
+    maxPos <- which.max(aRow)
+    newRow <- rep(FALSE, length(aRow))
+    newRow[maxPos] <- TRUE
+    return(aRow * newRow)
+}
```

Comment:
Seems like an internal function.

Reply:
And the same result you can get with a single line: `ifelse(x==max(x), x, 0)`, consider switching to this to keep it simple?